### PR TITLE
[DO NOT MERGE] - MapData POC

### PR DESCRIPTION
**Pull request by madadam**

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @afck

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/data/map_data.rs [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    sign_key: sign::PublicKey,
+    signature: Signature,
+    operation: PermissionOperation,
+}
+
+#[allow(unused)]
+pub enum PermissionOperation {
+    List,
+    Get {
+        // TODO: shouldn't this be `sign::PublicKey` ?
```
Comment (ustulation) [inline src/data/map_data.rs]: yes

File: src/data/map_data.rs [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+}
+
+// TODO: ordering based on name only. Verify it is sufficient!
+impl Ord for MapData {
+    fn cmp(&self, other: &MapData) -> Ordering {
+        self.name.cmp(&other.name)
+    }
+}
+
+// TODO: hashing based on name only. Verify is is sufficient!
```
Comment (afck) [inline src/data/map_data.rs]: I don't think it is: In the data manager, we'll probably exchange hashes in refresh messages to synchronise the state of a data chunk. For that, the `Hash` impl needs to take into account the data and owner fields.

Comment (afck) [inline src/data/map_data.rs]: Note that `BTreeMap` implements `Hash`, so with that, `Hash` could be derived.

**Comment by ustulation** [comment]
This is taxing the CI. It is better to not raise it as PR

**Pull request closed without merging**